CMakeLists.txt

```diff
@@ -855,3 +855,7 @@ else(NOT WIN32)
     legacy_stdio_definitions.lib Shlwapi kernel32 user32 gdi32 winspool comdlg32
     advapi32 shell32 ole32 oleaut32 uuid odbc32 odbccp32)
 endif (NOT WIN32)
+
+if (NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
+  message(FATAL_ERROR "Rippled requires a 64 bit operating system.")
+endif()
```

Comment:
According to this thread: http://cmake.3232098.n2.nabble.com/Detecting-32-64-bit-systems-td6019142.html  CMAKE_SIZEOF_VOID_P may not be reliable on OS X. 

"on Mac OS X if CMAKE_OSX_ARCHITECTURES is set to multiple architectures (especially, if they're mixed 32/64 bit) you can't rely on the value of CMAKE_SIZEOF_VOID_P."

I've read some alternatives similar to what @mellery451 suggests with option 2.